sklearn/metrics/_classification.py

```diff
@@ -2477,15 +2476,20 @@ def brier_score_loss(y_true, y_prob, sample_weight=None, pos_label=None):
         raise ValueError("y_prob contains values less than 0.")
 
     # if pos_label=None, when y_true is in {-1, 1} or {0, 1},
-    # pos_label is set to 1 (consistent with precision_recall_curve/roc_curve),
-    # otherwise pos_label is set to the greater label
-    # (different from precision_recall_curve/roc_curve,
-    # the purpose is to keep backward compatibility).
-    if pos_label is None:
-        if (np.array_equal(labels, [0]) or
-                np.array_equal(labels, [-1])):
-            pos_label = 1
-        else:
-            pos_label = y_true.max()
+    # pos_label is set to 1 (consistent with precision_recall_curve/roc_curve)
+    if (pos_label is None and (
+            _determine_key_type(labels) == 'str' or
+            not (np.array_equal(labels, [0, 1]) or
+                 np.array_equal(labels, [-1, 1]) or
+                 np.array_equal(labels, [0]) or
+                 np.array_equal(labels, [-1]) or
+                 np.array_equal(labels, [1])))):
+        raise ValueError("y_true takes value in {classes} and pos_label is "
+                         "not specified: either make y_true take integer "
+                         "value in {{0, 1}} or {{-1, 1}} or pass pos_label "
+                         "explicitly.".format(classes=labels))
```

Comment:
I think we want to keep the previous behaviour. The only addition is to raise an error when strings are passed and pos_lavel is None.

Reply:
The issue is that whether we want to regard larger integer as the positive label by default. I change here because I think it's strange, and it's also more consistent with other functions.